### DISCUSSION TITLE
MapGraphQL vs MapGraphQLHttp and MapGraphQLWebSocket

### DISCUSSION THREAD

**Discussion by RehanSaeed** [ChilliCream/hotchocolate#4287] [Q&A] [answered]
Is it possible to have `MapGraphQLHttp` and `MapGraphQLWebSocket` co-exist at `/graphql` just like `MapGraphQL` used to do? A lot of GraphQL tools prefer this. When I try it I get an exception:

```
Microsoft.AspNetCore.Routing.Matching.AmbiguousMatchException: The request matched multiple endpoints. Matches:

Hot Chocolate GraphQL HTTP Pipeline
Hot Chocolate GraphQL WebSocket Pipeline
```

In theory mapping them both to the same URL should be ok because the use different schemes.

- `https://localhost/graphql`
- `ws://localhost/graphql`

**Comment by OlegLviv** [comment]
any updates?

**Reply by tobias-tengler** [reply]
Not as far as I know, but I've properly raised an issue about it, so we can track it better :) #4707

**Reply by michaelstaib** [reply]
Not planned since you have `MapGraphQL` for it. We created `MapGraphQLHttp` and `MapGraphQLWebSocket` to explicitly not have them on the same route.

If you want to have them on the same route use `MapGraphQL` and pass in options for things you want to exclude.

**Reply by OlegLviv** [reply]
`         
builder.UseEndpoints(endpoints =>
            {
               endpoints.MapGraphQL<GraphQlSchema>();
            });
`
![image](https://user-images.githubusercontent.com/23611918/151787530-5fbe180c-b520-4341-bbc2-5241e5ae1ad0.png)
![image](https://user-images.githubusercontent.com/23611918/151787589-b5455b8d-13bc-4eac-aed1-18c0ea6d2629.png)
My front-end app can't establish a connection through ws because I have not registered the route so I need to separate them 
but before I updated GraphQL NuGet packages it works... how?

**Reply by michaelstaib** [reply]
If you just use `MapGraphQL` everything will be configured on a single endpoint.

**Comment by michaelstaib** [comment, marked as answer]
Not planned since you have `MapGraphQL` for it. We created `MapGraphQLHttp` and `MapGraphQLWebSocket` to explicitly not have them on the same route.

If you want to have them on the same route use `MapGraphQL` and pass in options for things you want to exclude.
